packages/blockchain-utils-validation/src/blockchains/tezos.ts

```diff
@@ -0,0 +1,64 @@
+import { AccountID } from 'caip'
+import { RemoteSigner } from '@taquito/remote-signer'
+import { char2Bytes } from '@taquito/utils'
+import { BlockchainHandler } from '../blockchain-handler'
+import { LinkProof } from '@ceramicnetwork/blockchain-utils-linking'
+import axios from 'axios'
+
+export const ADDRESS_NOT_FOUND_ERROR = new Error(`Address not found on the Tezos blockchain`)
+export const PUBLIC_KEY_NOT_PUBLISHED_ERROR = new Error(
+  `Public key not published to the Tezos blockchain`
+)
+
+const namespace = 'tezos'
+
+const FAKE_API_ENDPOINT = 'https://fake-api.example.com' // this enpoint is never actually called
+
+/**
+ * creates a function which attempts to resolve the public key associated with the wallet address
+ *
+ * @param {string} address - a Tezos wallet address to lookup
+ * @returns {() => Promise<string>} - the fetching function for the public key of the address
+ * @throws {Error} - if the address is not found on the blockchain
+ * @throws {Error} - if the public key is not published to the blockchain
+ */
+function publicKeyFinder(address: string): () => Promise<string> {
+  // request the public key from the Tezos blockchain
+  const request = axios.get(`https://api.tzstats.com/explorer/account/${address}`).catch(() => {
+    throw ADDRESS_NOT_FOUND_ERROR
+  })
+  return async (): Promise<string> => {
+    const result = (await (await request).data)?.pubkey
+    if (result) {
+      return result
+    } else {
+      throw PUBLIC_KEY_NOT_PUBLISHED_ERROR
+    }
+  }
+}
+
+// can only validate proofs with wallet addresses (public key hashes) that have:
+// - at least one transaction on the blockchain
+// - their public key published to the blockchain
```

Comment:
Why is this needed? Seems wrong. With all other blockchains you can verify signatures on just the address (public key hash) because we're using recoverable signatures. Is Tezos really not doing the same (I'm pretty sure it has to!)?

Reply:
No other blockchain hashes the wallet's public key to generate the wallet address. Other blockchains just encode the public key as a string. So unless SHA2-256 is broken, the public key has to be published somewhere.

Reply:
> No other blockchain hashes the wallet's public key to generate the wallet address.

Yes they all do 😄

Reply:
See for example: https://www.quora.com/What-encoding-is-used-for-Ethereum-addresses?share=1 for ethereum

Reply:
> > No other blockchain hashes the wallet's public key to generate the wallet address.
> 
> Yes they all do 😄 

Whoops

Reply:
> See for example: https://www.quora.com/What-encoding-is-used-for-Ethereum-addresses?share=1 for ethereum

This is mapping from public key to hash tho. We need to map from hash to public key, and that still requires an on chain lookup.

> You can if and only if a transaction has been sent from the account. When you send a tx, you sign the transaction and it includes these v r and s values. You parse these from the signed tx and then pass these v r and s values and the hash of the transaction back into a function and it'll spit out the public key. This is actually how you get the from address of a transaction.
Source: [Ethereum StackExchange](https://ethereum.stackexchange.com/questions/13778/get-public-key-of-any-ethereum-account)

Tezos just has the public key directly in the transaction since it supports multiple types of keys.

Reply:
> and that still requires an on chain lookup

No this is not the case. If you have the message and the signature, you can recover the public key, then hash it to see if it matches the address.